src/core/qgsauxiliarystorage.h

```diff
@@ -29,6 +30,60 @@
 class QgsProject;
 
 /**
+ * \class QgsAuxiliaryField
+ *
+ * \ingroup core
+ *
+ * \brief Class allowing to manage fields from of auxiliary layers
+ *
+ * \since QGIS 3.0
+ */
+class CORE_EXPORT QgsAuxiliaryField : public QgsField
+{
+  public:
+
+    /**
+     * Constructor
+     *
+     * \param def Definition of the property to be stored by this auxiliary
+     *  field.
+     */
+    QgsAuxiliaryField( const QgsPropertyDefinition &def );
+
+    /**
+     * Destructor
+     */
+    virtual ~QgsAuxiliaryField() = default;
+
+    /**
+     * Returns the property definition corresponding to this field.
+     */
+    QgsPropertyDefinition propertyDefinition() const;
+
+    /**
+     * Returns the name of the field.
+     */
+    using QgsField::name SIP_SKIP;
+
+    /**
+     * Returns the name of the auxiliary field for a property definition.
+     *
+     * \returns def The property definition
+     * \returns joined The join prefix is tok into account if true
+     */
+    static QString name( const QgsPropertyDefinition &def, bool joined = false );
```

Comment:
This is messy - why not just rename this method as nameFromProperty and avoid the using QgsField::name requirement?

Reply:
You're right, done!